[#10195] Separate keystore dir per multiacc

rasom opened the pull request:
#10195

Having all keys related to specific multiacc in a separate dir will simplify removing of the multiacc.
Also it will allow adding of the same account to different multiaccs.

depends on https://github.com/status-im/status-go/pull/1993

status: ready

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 5eb08115 | [#1](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-10847/1/) | 2020-06-18 10:48:01 | ~11 min | `android-e2e` | [:package:`apk`](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-200618-103629-5eb081-x86.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-prs.ams3.digitaloceanspaces.com%2FStatusIm-200618-103629-5eb081-x86.apk)|

hesterbruikman commented:
fyi @johnlea-quiup as this relates to key management, specifically removing a 'multiaccount' (master keypair)

yenda commented on src/status_im/multiaccounts/create/core.cljs:
```diff
@@ -71,6 +71,12 @@
 (fx/defn create-multiaccount
   [{:keys [db]}]
   (let [{:keys [selected-id key-code]} (:intro-wizard db)
+        key-uid (reduce
```
unless I'm misunderstanding, it seems like `(first (filter f coll))` would be more idiomatic here

rasom replied:
that would be `(:key-uid (first (filter f coll)))`, but why it would be more idiomatic?

rasom replied:
btw then probably `(:key-uid (some f coll))` would make it with less parentheses. Idk how idiomatic it is `¯\_(ツ)_/¯`

rasom replied:
oh well it would be just `(some f coll)`, that should be a winner

oskarth commented:
Small request - could you please rebase to latest master so this commit https://github.com/status-im/status-react/commit/b5fda1234e5447cbc626059dfc78fcfbabc9e926 gets included? This means we aren't running E2E tests on eth.prod cluster anymore, and since this happens on every push, the impact is quite big. This ensures we have more accurate metrics going forward from Jul 1 onward, which would be awesome. See https://discuss.status.im/t/user-growth-and-retention/1782 for more

status-im-auto commented:
## 99% of end-end tests have passed
```
Total executed tests: 94
Failed tests: 1
Passed tests: 93
```
<h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534466">test_keycard_fetching_balance_after_offline</a></b></td></tr><tr><td><p><blockquote><div>Transaction #5, amount is 0.002101327</div><div>{'blockNumber': '8212419', 'timeStamp': '1593684851', 'hash': '0xeaeccd60e83509b4626c07d52179894f60f675ce6d9707b434429e22fd16cdbf', 'nonce': '1232', 'blockHash': '0x62c47075b490b368e68dc4150cf33d07caad80ca56026ba0d160bc14e18c27af', 'from': '0xc8f8b3376814d0dfe61e4b9eaf7970697c24f6b2', 'contractAddress': '0xc55cf4b03948d7ebc8b9e8bad92643703811d162', 'to': '0xf184747445c3b85ceb147dfb136067cb93d95f1d', 'value': '2101327000000000', 'tokenName': 'Status Test Token', 'tokenSymbol': 'STT', 'tokenDecimal': '18', 'transactionIndex': '6', 'gas': '131542', 'gasPrice': '1000000000', 'gasUsed': '109619', 'cumulativeGasUsed': '614142', 'input': 'deprecated', 'confirmations': '508'}</div></blockquote></p><code> Transaction with amount 0.002115921 is not found in list of transactions, address is f184747445c3B85CEb147DfB136067CB93d95F1D</code><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/327b9068a2e44e4f96a2a41935a09dbc?auth=a9d3a3f6d1fbc184f4032e7711e89bb6">Steps, video, logs</a></li><li><a href="https://assets.saucelabs.com/jobs/327b9068a2e44e4f96a2a41935a09dbc/0056screenshot.png?auth=a9d3a3f6d1fbc184f4032e7711e89bb6">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (93)</h3><details><summary>Click to expand</summary><br/><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534455">test_decline_transactions_in_1_1_chat</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/d77154f32d7445889b277f8ba58bc61a?auth=e600dc36ed5393421ebb5ba419a97a94">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/364955f499ca48088417805af43c8657?auth=7ace48c69ffef66cca37bb28f628783a">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534401">test_delete_chats_via_delete_button</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f47fd4e7c68d41d5bda793d660c36678?auth=f9bacc31a7d28f9c3cd59f748a304e6e">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534361">test_can_see_all_transactions_in_history</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/a4a19d82754845d49edff4292032fb3d?auth=ed596c4425cb5ae48cccdc46d3cf5590">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534431">test_install_pack_and_send_sticker</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/2bcc579ec6ce4b8b82744bd93c9a2e11?auth=9c0dcecfa47835ebd4d676e78fc4a12d">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f663b123bf524401826c9d58929e7957?auth=424095d3f6a203f936cbe02a070f3e52">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534359">test_login_with_new_account</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/8c8576d6eea241bd8ddb25ddc723667d?auth=fa36e8ffbca248a846f918a91a7ee554">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534399">test_timestamp_in_chats</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/bdf50c2d373b4af08f0fdece0dd80258?auth=b9ae49ab9415869f7f335942675e333e">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/9a63f244e1a14c288d844b80bf28bca1?auth=f38f2b82739aa4e5c293e2c0c130f745">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534414">test_password_in_logcat_sign_in</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/15a8d65a252144948c6053f823ad7fba?auth=2bacbe6cd5f0a7d2546732ae92a33d51">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534391">test_logcat_recovering_account</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/c79433950c6c42f09cc0522567a68aaf?auth=a73636a39ffbf04b3bd3cda00846eb0c">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534400">test_public_chat_clear_history</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/dd95e4a4b4bb4001b5cdd9b9671f8c7a?auth=bde45a032329e411496e3bd29a8c33f9">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/5d4a8117e639414dbfdb11a02eb7bb04?auth=99a0909f7cd3ff0fd5ade00ae975cccd">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>10. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534403">test_need_help_section</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/fd36057670814bf99275b6f00979f784?auth=69d4eab3d397278bb80d43b9bf2600db">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>11. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534393">test_send_and_open_links</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/83c8e041d4364105a2cb9f97797ad155?auth=ed28f9e5b626b8545afd7ed3878344be">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/a370def5a83b426c8d9253a9b2c3b3f3?auth=21083ffe694ed071db4c270032e33230">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>12. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534444">test_mobile_data_usage_popup_stop_syncing</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/0e1fa36d54e34cd6881d0b72c7f1b12c?auth=b1b1937c83dc0b4b8b4874afe2a61908">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>13. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534445">test_mobile_data_usage_settings</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/7878171a96724c29a733366766a5f50a?auth=a1e1905f60dc3e0cd564ebaaba7ff87f">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>14. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534350">test_create_new_group_chat</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/d27f108dda6249fc82c7ae0d71b39497?auth=e962558885a24497c95dc58522a0e612">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/4a9edad5d2b74cbf9c706bb406e4c327?auth=db3a877d942e8c5af0c99030b905c0e6">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>15. test_sign_typed_message</b> (TestRail link is not found)</td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/103042c8b7d34b28a6ef156493937083?auth=7c4f61b16d259bb38639c62565020c30">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>16. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534352">test_user_can_switch_network</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/5940ff6df3ae41d3b6a37e29aad6c179?auth=3a3228f2272ebd751b4175ed9cf4894e">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>17. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534424">test_mobile_data_usage_popup_continue_syncing</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/8b41ad37b4f54914b3d73c56eec88b54?auth=aef25eb10af166f6d423d7748171caec">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>18. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534395">test_home_view</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f410016611ad47a8a2d504a202be2765?auth=99dc524a329b361566ea362d2e6d6641">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>19. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534366">test_open_google_com_via_open_dapp</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3e897f2b37a54570ae8652744503a6a8?auth=0deb755a02a6e510faac6ae349d2588e">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>20. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534368">test_share_contact_code_and_wallet_address</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/25b4442d886a456fba7ecdafd65abc1a?auth=c6986ef7e7a700b731c20cc47274110d">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>21. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534408">test_connection_is_secure</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/abfeb6928754431aad129bee90ea39b4?auth=77085a04309f6980f5bcb19df76e3498">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>22. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534363">test_add_to_contacts</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/054d6c5b3f2048eb95273a74a4922a78?auth=1aa9526b87d50bec1e21d204a6cc6e68">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/8a1f178a38114f70b37c65d82e44f103?auth=77141f3cc1563e01ad8278375c000550">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>23. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534457">test_add_account_to_multiaccount_instance_private_key</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/fbbeec9b3d084d4aa4b09c9b23ca416e?auth=0b68e2331e1e71577ca658328805cb34">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>24. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534383">test_refresh_button_browsing_app_webview</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/baf1892eef134bf28d9175979d4be855?auth=2c4a32b177d7da714b64814a18ba1a6c">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>25. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534426">test_keycard_send_two_transactions_one_after_another_in_dapp</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/e866cec5748a4dffb8498f46189a58db?auth=f2c2594aaba1b7c24b8a5fa5ef5be71a">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>26. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534417">test_can_add_existing_ens</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/097cec863d7d4fe1a6311c08f14ae92b?auth=97072d6243502bb41d98cae78ee6ba42">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>27. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534411">test_transaction_wrong_password_wallet</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/37cc6212cf92435b87d0fc6ff5949b19?auth=3cdea927a33ff928d453152c78977ff2">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>28. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534360">test_public_chat_messaging</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/817ffa7468d242938c2ea30dfcd94bc3?auth=eb24cf5d1c88d5e6c719c7d8fd261bd8">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/d8193dd906a64570970ce8933e8db1f2?auth=27149e727cbf36df0336cc5ca0e20667">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>29. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534386">test_unread_messages_counter_public_chat</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/df4a5e490d7c4802b6d12e4325f9cc3b?auth=1c1d555862cc307d89a0d21c14f57501">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/9ac67d216d4945e4903202af2071c3ac?auth=d18d26d16ff53fe81388840e004980f6">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>30. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534465">test_keycard_send_eth_from_wallet_to_address</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/3ed41fb1657f41b6a4c1a68fd0c44e3a?auth=15ce57ecf4b143f32a66913b07fbdeb7">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>31. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534449">test_keycard_send_transaction_from_daap</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/81b1826351cb403989c110ed59fab5bc?auth=14a5fc25cc2eadc899630504e8bbc175">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>32. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534355">test_text_message_1_1_chat</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/7e81f90bd0174a9987bb6396c4e99466?auth=835495f1cb63f9e8f9c9e70b58a0fdcf">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/bd158421a12d411b9b2841cf20b96a93?auth=d949452c95f07fc3233153e30c83dc33">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>33. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534432">test_can_use_purchased_stickers_on_recovered_account</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/68a642c7c2b54ad2b01cecf193b40bac?auth=f3b6347e4a94f74e825bd0570f190ce0">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>34. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534398">test_open_transaction_on_etherscan</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/6130cae393f24054b787a601463daa50?auth=4d118533e0fb72d2bcdf0822f099fe26">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>35. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534420">test_onboarding_screen_when_requesting_tokens_for_recovered_account</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/52eb3f52f1cb49989abf6e1366675ff8?auth=c564b52fc8003b8140df46619554ad40">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>36. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534397">test_contact_profile_view</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/b035b8dc5ac643bc9ffac1dc1b53993c?auth=b3e45e8f6aebad59763b69ab780f99ea">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>37. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534351">test_offline_add_new_group_chat_member</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/905a90af1ed7413eaaebc3ca415d3ae7?auth=2fb44ed47d3b1edbed66c5c7e171dd86">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/300798a83f48434cb332ea2d1adcd9e0?auth=f800396679a263b145edf3f5a3b60997">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://saucelabs.com/jobs/d71dc8fd1c5b4e0890d2554268509b79?auth=b1873ae4035bdf4e1430b883c6aa02e0">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>38. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534415">test_logcat_backup_recovery_phrase</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/8a47201c340a4bc980fdb3337343194a?auth=34bfc936b6ab0e4a85c146667e59c78d">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>39. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534409">test_start_chat_with_ens</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/468bbed38ec64ac097e4fa1b64068b97?auth=511b4e7b2af2de041a47c4eaab158886">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>40. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534372">test_add_and_remove_contact_from_public_chat</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/2614b8374ff343afb27ed2e3dcf8c240?auth=db588b2bfc173096bd79966f6df364d9">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/a3346ee173be4a2f9e54dbf062429bad?auth=0caefb7598a529b3bffcc51f369ba802">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>41. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534456">test_add_account_to_multiaccount_instance_seed_phrase</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/ee6ef350e3124f2f9a0b132e2031e518?auth=5eb6d68a84ea24235deb49e6f12c378f">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>42. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534437">test_open_blocked_site</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/0bd8a4981b4149eb8236b8f6b4829a5c?auth=cf0c04de3b1812f6658e1671efbb0bfd">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>43. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534354">test_open_chat_by_pasting_public_key</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/ec8fad746b534590aa902439ea6a9cfc?auth=5b6cc5a21c96f24314a8462ab47b3e81">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>44. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534429">test_pair_devices_sync_one_to_one_contacts_public_chat</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/58d905b2c00849f0ac46e6017d0ea978?auth=a8f5307e81f938472c8a522773a9f03b">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/feaa7aa19cde4178b98a3317ab16260d?auth=610adaf7d71c3add5e3e68558da4b8d6">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>45. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534451">test_keycard_sign_message_from_daap</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/2187274e5f1549a28ef74c8b59e1850a?auth=25fed8706bac93719fc6d1ec49f2472f">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>46. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534362">test_send_non_english_message_to_newly_added_contact</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/691844843b944be9824247a742a1b23d?auth=5ac84e2e2cdc05fad7aec587ab615bfc">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/bce7bd8c6c4846dba31097fa5ddae0ae?auth=48f8e33d100af43e662d23093a31ebd8">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>47. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534410">test_ens_username_recipient</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/e8768f703dd5483ca71a250cffd5d713?auth=e0dd9f709ee9ab743c2dec50dce1fb4c">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>48. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534434">test_fetch_more_history_in_empty_chat</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/167099e6d28540ac8b09409fb7c231e2?auth=ab51d1968015f482f441b1109b03d0f3">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>49. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534406">test_open_public_chat_using_deep_link</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/fbbcedc6482f4b5689e97086f6f6db26?auth=9d3cfc0dbd3245075947995a8909b22d">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>50. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534384">test_switch_users_and_add_new_account</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/6979a8d6e15b44acaad9e4aefc0509d3?auth=572b62f64713b474b8831865145133d6">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>51. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534441">test_add_account_to_multiaccount_instance_generate_new</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/2cdf33800a4b4d159fc6915091011925?auth=e96d3e2de05b8bc99af824b6337e3b44">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>52. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534394">test_copy_contact_code_and_wallet_address</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/ef4626495b634156b90b2d0901de9c46?auth=34da4cbc823ac8c3630af02976060c07">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>53. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534357">test_send_transaction_from_daap</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/5de7961dfe984bbf8c5040b86a5682c9?auth=0ebb25cb6a310b4b15c518e1d9211293">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>54. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534426">test_send_two_transactions_one_after_another_in_dapp</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/2635ed00598944c1845ae2f1bbeb04d8?auth=263047a5db11c14079205a14d0027b16">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>55. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534472">test_image_in_one_to_one_send_save_reply</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/494047517f934857b06257e55ea72029?auth=4f2621007366b8b6419542a45d670734">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/9b108bde5ea94a3caed8fa2bfc472db9?auth=c3e9600d1703cf5b97bc85aa7ded7c1e">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>56. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534421">test_add_new_keycard_account_and_login</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/b9f57e705a1649c394491e9d24ff4425?auth=297a9720c7161ed4b853b433b31ecfb0">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>57. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534402">test_long_press_delete_clear_all_dapps</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/2ddc5265687e4f59b810ab43b45585a7?auth=e6b2605f4a0bb316975bfbec65e79db0">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>58. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534364">test_copy_and_paste_messages</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/6fb894e1af66443d838e18f1630a761f?auth=eb1b33cbf2d485956d9d0c59faccc729">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>59. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534367">test_back_forward_buttons_browsing_website</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/b4950127c9724269b2fd29e6e3ecc780?auth=4b14a6ea35224257e227e44e8f1131c4">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>60. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534413">test_password_in_logcat_creating_account</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cba8e4151e924b1c820192fbe17222a1?auth=fbb4ead6b243e197a16c481566537846">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>61. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534379">test_sign_message_from_daap</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/4bbf28c5bc764f65b13f99307c8b7c8b?auth=7a98ad4500e48ac36f3092c437e423c2">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>62. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534423">test_dapps_permissions</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/b446f00d43a54aa08228171c8963966c?auth=19f500b32984d79d5e5ec35e25c71672">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>63. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534407">test_request_public_key_status_test_daap</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/9db6cf3c5edd4e0db731a784c7aa5caf?auth=1943659480236ee57e9a1ae35d1cffbd">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>64. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534365">test_long_press_to_delete_chat</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/4a955d3ddcf347f095f483b3eef34cc6?auth=5dd2510ee74679e16eb26eec211bac08">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>65. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534392">test_log_level_and_fleet</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/5d58426ec29349ea8ce3a4b0b61c493e?auth=4cec9f10f872f960a22692a145287d04">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>66. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534448">test_add_and_delete_watch_only_account_to_multiaccount_instance</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/56aa2ab74bff44b3af4173081ec7936c?auth=f3947f775a70ddf2d4c4b544f5412e8a">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>67. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534471">test_recover_account_from_new_user_seedphrase</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/44372525af0b47b9812a5bc4f79cc753?auth=7988c778ff37d167947bd577243cab74">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>68. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534435">test_send_transaction_with_custom_token</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/8efbeed8afd44f019d4b2d5cc917401e?auth=70a243356ea6196e2fb65b1610bfb714">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>69. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534452">test_send_eth_in_1_1_chat</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/24dbc8a042224ff4992a69b57db5f662?auth=cb66bec84e2bfaed3f369b8a8d89e029">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/ebd6e8e874a14e259e44df656f5a1880?auth=a03ba7d8639aa864fcbcda0e380bb39e">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>70. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534425">test_send_two_transactions_in_batch_in_dapp</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/b281a78042dc403dbb0b3b8fb480fccf?auth=15d29924a0427ca8d45c84be6669df36">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>71. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534442">test_ens_in_public_and_1_1_chats</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/9a2ebd7a2fd341bfaa6d20943918b308?auth=f716c2c277118b3bbd531a66d0eed424">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/ef29147aedaf4358baf8802171dfe01d?auth=4d08ac23949f789f7c022f569c1718d0">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>72. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534433">test_block_user_from_public_chat</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/7e5b54fb4ec1421b8194d263336f5053?auth=a9124fcdb69e3257cd3cf504efb4075f">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/02e7b82916304b51a1d733134c289665?auth=7fe530a0943d73e12a58090972e507d8">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>73. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534446">test_fetching_balance_after_offline</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/647b09c60dde448b88a4a324b72e8373?auth=da8a9992c6ab117ca35e8a2fa2bcbf0e">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>74. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534369">test_offline_status</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/6e2c3fe4b14946bd977e6616e54920c4?auth=4ba74deb2d29fe05273b103fb4cecb25">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>75. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534373">test_wallet_set_up</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/97282bb12e924957a730e58bb4ef4c47?auth=afdc3d18fad50c77ad4cc422085ae99d">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>76. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534378">test_manage_assets</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/c6f9e9c71603454caf600102aa313690?auth=820a064ec7f6fbea33ac841960f5436e">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>77. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534358">test_offline_messaging_1_1_chat</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/98c524da687b4e88b8aafaa441a8a525?auth=c1ffda0936396840d38d919b6768f940">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/0ef0bb2574ef447d9a60b7ec610cca67?auth=69acfc9b6f1b0e9e95c44b36de278471">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>78. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534404">test_account_recovery_with_uppercase_recovery_phrase</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/2908bccc36a44c2a8d4b43872268bab9?auth=bbaa7b308f3191fdb975f4fa97a5646f">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>79. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534381">test_send_token_with_7_decimals</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/954ce57a79574750a89b4e5358f8be25?auth=df7ab1cf9461b972ed4f0ab51160daca">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>80. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534396">test_user_can_see_all_own_assets_after_account_recovering</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/92bb5418afbe4ae2aa68e07421ba0372?auth=f6c713f9c3a2b0bd36e9a3612e5321af">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>81. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534388">test_pass_phrase_validation</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/26b9e6d0b1c8454c84b9bbcebe1c124d?auth=cb99f433e8fc799e15ce431407d63422">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>82. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534467">test_keycard_can_see_all_transactions_in_history</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/12d58b07d52b4991a31ad381bd28bd5b?auth=8ffbbfe905b37e003abb177631ece729">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>83. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534463">test_keycard_send_two_transactions_in_batch_in_dapp</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/4270f0d41f104524aac08b51aa9f7b3c?auth=63a419e2878b2da9f9c7c9879dce4bef">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>84. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534370">test_send_emoji</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/33f0fedafcd24b96bbed4f25cbcbcd4f?auth=e233da7e15b54688b0c4c76c06708ebd">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>85. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534382">test_filters_from_daap</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/d6f552b298834531a4ba9c9ad2ab8359?auth=7220515d8eaa5e786ea7f899aa6e0df1">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>86. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534419">test_redirect_to_public_chat_tapping_tag_message</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/eeedfa1069544ce49aa28a2adefce5d1?auth=edef51b866082ef39748f3b8851a84c6">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>87. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534447">test_restore_account_from_mnemonic_to_keycard</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/0ec7b563db6c4a8382e740b1f3b2675d?auth=cf97a9f74e458bf92d56fd584e468027">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>88. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534459">test_send_eth_to_ens_in_chat</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/eea2f44725604849a4f7939a63a93f59?auth=bad5a5ce05cee5d8678927c7a1230b16">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>89. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534356">test_send_eth_from_wallet_to_address</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/76ce4d799afc44c3a159fd15f18b0f01?auth=44a04899fd16da78443c1e2db0b20476">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>90. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534454">test_request_and_receive_stt_in_1_1_chat_offline</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/0d05f7b92d254ba8bc1e4d04b3c2162e?auth=973c4798ac2dab652a702f85d90f906e">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/4f08b67794334c8e963d6c2ca47d36ae?auth=6808c466868c759d27a54563a7f1af33">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>91. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534375">test_messaging_in_different_networks</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f027acfabd9f4886b64791dd3d828bf5?auth=0c1206385ad0ebbb0fd80041fb8bd83c">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/05212acbbe574f56ae7877eca2e06d69?auth=ee510f3f4e121c0b2b247717972e7ced">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>92. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534380">test_collectible_from_wallet_opens_in_browser_view</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/2f3500654e4f409f9aae5fb2a16a0c16?auth=b3cea870b2234f0d00c9c87f433dc5f4">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>93. <a href="https://ethstatus.testrail.net/index.php?/tests/view/1534412">test_insufficient_funds_wallet_positive_balance</a></b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/ec7ec159d6584bfe9089bb0010e48c31?auth=7196179d10f3b72ebea21b1a325038c3">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

churik commented:
## Can't send transaction from account if it is added as separate multiaccount with keycard

**Steps:**
- user A: restore multiaccount A from seed phrase A (choose keycard Storage), login
- user A: go to wallet > Add account > Enter seed phrase B > Confirm, log out 
- user B: restore multiaccount from seed phrase B, log out
- User A: login to multiaccount A
- User A: try to send transaction from account B

**Expected result:** can send Transaction
**Actual result:** 
![IMAGE 2020-07-03 17:36:03](https://user-images.githubusercontent.com/4557972/86483136-ab2cda00-bd53-11ea-8dc4-718486d8d133.jpg)
[Status-debug-logs.zip](https://github.com/status-im/status-react/files/4870433/Status-debug-logs.zip)

rasom commented:
> user A: restore multiaccount A from seed phrase A (choose keycard Storage), login
@churik is this even possible?

churik commented:
Previous issue is not reproducible with the same steps, but I've got another error:

**Steps:**
1) user A: restore multiaccount A from seed phrase A (choose keycard Storage), login
2) user B: create account > go to wallet > Add account > Enter seed phrase A > Confirm, log out 
3) User A: login to multiaccount A > send transaction > log out
4) User B: login to multiaccount B > try to send transaction from added on 2 step account 

**Expected result:** can send Transaction
**Actual result:**
![IMAGE 2020-07-06 13:36:14](https://user-images.githubusercontent.com/4557972/86589283-a9e6f180-bf8d-11ea-84f2-1d727df6256a.jpg)

Logs: [Status-debug-logs.zip](https://github.com/status-im/status-react/files/4878614/Status-debug-logs.zip)

churik commented:
@rasom 
### 'Cannot retrieve a valid key for a given multiaccount error' after upgrade with 2 multiaccounts (1 of them is keycard)
**Steps:**
1) installed Status Pr for upgrade (used from https://github.com/status-im/status-react/pull/10686: https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-200706-103428-6df9a2-pr-universal.apk)
2) recover Keycard multiaccount 
3) log out
4) create ordinary multiaccount
5) upgrade to  PR build (https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-200706-131420-e9de81-pr-universal.apk)
6) try to login to ordinary multiaccount

**Expected result:**
logged in successfully

**Actual result:** error `Cannot retrieve a valid key for a given account: no key for given address or file"`
If after that login to keycard account, you'll see wallet/profile name of ordinary multiaccount inside keycard multiaccount
[Status-debug-logs.zip](https://github.com/status-im/status-react/files/4885176/Status-debug-logs.zip)

churik commented:
Tested on Android 10:
- upgrade with several multiaccounts (one of them was keycard with several accounts inside)
- added account from seed phrase to ordinary account (already opened as keycard multiaccount)
- checked transactions from added accounts

Tested on IOS 13:
- "Account already exists" (when add account from seed phrase within same multiccount or adding the same account twice)
- upgrade with several multiaccounts + enabled "Save my password"
- added account from seed phrase to ordinary account (already opened as keycard multiaccount) and vise versa
- checked transactions from added accounts.

Looks good to me,  thank you for patience @rasom

rasom committed:
[#10195] Separate keystore dir per multiacc

Having all keys related to specific multiacc in a separate dir will
simplify removing of the multiacc. Also it will allow adding of the same
account to different multiaccs.
